akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala

```diff
@@ -124,17 +124,19 @@ private[remote] class ArteryTcpTransport(_system: ExtendedActorSystem, _provider
     def connectionFlowWithRestart: Flow[ByteString, ByteString, NotUsed] = {
       val flowFactory = () ⇒ {
 
-        afr.loFreq(
-          TcpOutbound_Connected,
-          s"${outboundContext.remoteAddress.host.get}:${outboundContext.remoteAddress.port.get} " +
-            s"/ ${streamName(streamId)}")
-
-        // FIXME use the Flow.lazyInit from https://github.com/akka/akka/pull/24527
-
         val flow =
           Flow[ByteString]
-            .prepend(Source.single(TcpFraming.encodeConnectionHeader(streamId)))
-            .via(connectionFlow)
+            .via(Flow.lazyInit(_ ⇒ {
```

Comment:
the whole thing is wrapped in `RestartFlow.withBackoff` and then I don't want to open new connection immediately after backoff unless there is a new message

cc @gosubpl this was what I needed it for in Artery

Reply:
cool! my use case is a bit different, similar to `switchMap`. I use it in an HTTP route, that has elements and commands. Commands result in different processing graphs (flows) for elements. So I `splitWhen` on each command, `Flow.lazyInit` build processor for subsequent elements and `concatSubstreams` finally.